tests/functional/adapter/materialization/test_incremental_delete_insert.py

```diff
@@ -0,0 +1,160 @@
+import pytest
+
+from dbt.tests.adapter.incremental.test_incremental_unique_id import BaseIncrementalUniqueKey, seeds__seed_csv, \
+    models__trinary_unique_key_list_sql, models__nontyped_trinary_unique_key_list_sql, models__unary_unique_key_list_sql, \
+    models__not_found_unique_key_sql, models__empty_unique_key_list_sql, models__no_unique_key_sql, \
+    models__empty_str_unique_key_sql, models__str_unique_key_sql, models__duplicated_unary_unique_key_list_sql, \
+    models__not_found_unique_key_list_sql, models__expected__one_str__overwrite_sql, \
+    models__empty_str_unique_key_sql, models__duplicated_unary_unique_key_list_sql
+
+
+seeds__duplicate_insert_sql = """
+-- Insert statement which when applied to seed.csv triggers the inplace
+--   overwrite strategy of incremental models. Seed and incremental model
+--   diverge.
+
+-- insert new row, which should not be in incremental model
+--  with primary or first three columns unique
+insert into {schema}.seed
+    (state, county, city, last_visit_date)
+values ('CT','Hartford','Hartford',DATE '2022-02-14');
+
+"""
+
+seeds__add_new_rows_sql = """
+-- Insert statement which when applied to seed.csv sees incremental model
+--   grow in size while not (necessarily) diverging from the seed itself.
+
+-- insert two new rows, both of which should be in incremental model
+--   with any unique columns
+insert into {schema}.seed
+    (state, county, city, last_visit_date)
+values ('WA','King','Seattle',DATE '2022-02-01');
+
+insert into {schema}.seed
+    (state, county, city, last_visit_date)
+values ('CA','Los Angeles','Los Angeles',DATE '2022-02-01');
+
+"""
+
+models__expected__one_str__overwrite_sql = """
+{{
+    config(
+        materialized='table'
+    )
+}}
+
+select
+    'CT' as state,
+    'Hartford' as county,
```

Comment:
nit: can be done with `VALUES ... AS t(....)` as well

Reply:
This is an override from the original test in dbt's code. I prefer to keep it as close to the original as possible in case we need to compare it later on.